### ISSUE TITLE
False Positive `empty_enum_arguments`

### ISSUE THREAD

**Issue by utahwithak** [realm/SwiftLint#3883]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
I'm getting a false `empty_enum_arguments` on this line:
``` swift
guard case let .success(tracksTableResult) = result, !newOperationGroup.isCanceled() else {
```
Only on the Xcode 13.3 beta though, it doesn't trigger on Xcode 13.2. Worth mentioning, `isCanceled` is a function, not a var, definitely not an enum. 

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.46.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Paste your configuration file:

```yml
opt_in_rules:
    - nslocalizedstring_require_bundle

disabled_rules:
    - cyclomatic_complexity
    - discarded_notification_center_observer
    - identifier_name
    - implicit_getter
    - file_length
    - force_cast
    - force_try
    - function_body_length
    - function_parameter_count
    - large_tuple
    - legacy_constructor
    - line_length
    - mark
    - nesting
    - notification_center_detachment
    - todo
    - type_body_length
    - weak_delegate
    - fallthrough
    - block_based_kvo
    - xctfail_message

custom_rules:
  backgroundtask_usage:
    name: begin/endbackgroundTask Usage
    message: Do not use beginBackgroundTask or endBackgroundTask. Instead, use the BackgroundTask wrapper in Utilities.
    regex: beginBackgroundTask\(|endBackgroundTask\(
    severity: error
  apply_usage:
    name: DiffableDataSource Apply Usage
    message: Do not use apply. iOS 14 and iOS 15 have different behaviors when using apply with no animation. Use sf_apply.
    regex: (?<!sf_)apply\([^\n]*animatingDifferences
    severity: error
  id_usage:
    name: ID Suffix
    message: Identifier, arguments, parameters should not use mixed case for ID.
    regex: Ids?\b
    match_kinds: 
      - identifier
      - argument
    severity: warning
  url_usage:
    name: URL Suffix
    message: Identifier, arguments, parameters should not use mixed case for URL.
    regex: Urls?\b
    match_kinds:
      - identifier
      - argument
    severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? no
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 13.3 Build version 13E5104
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
guard case let .success(tracksTableResult) = result, !newOperationGroup.isCanceled() else {
```
With Xcode version 13.3

**Comment by marcelofabri** [comment]
This should be fixed in [0.46.4](https://github.com/realm/SwiftLint/releases/tag/0.46.4) (https://github.com/realm/SwiftLint/issues/3850)
